assembl/static2/js/app/reducers/adminReducer/voteSession.js

```diff
@@ -73,4 +82,116 @@ const voteSession: VoteSessionReducer = (state = initialState, action) => {
   }
 };
 
-export default voteSession;
\ No newline at end of file
+export const modulesInOrder = (state: List<number> = List(), action: ReduxAction<Action>) => {
+  switch (action.type) {
+  case UPDATE_VOTE_MODULES:
+    return List(action.voteModules.map(m => m.id));
+  case CREATE_TOKEN_VOTE_MODULE:
+    return state.push(action.id);
+  case DELETE_TOKEN_VOTE_MODULE: {
+    const index = state.indexOf(action.id);
+    let newState = state;
+    newState = newState.delete(index);
+    return newState;
+  }
+  default:
+    return state;
+  }
+};
+
+const defaultTokenModule = Map({
+  type: 'tokens',
+  titleEntries: List(),
+  instructionsEntries: List(),
+  exclusive: false,
+  tokenTypes: List()
+});
+
+export const modulesById = (state: Map<string, Map> = Map(), action: ReduxAction<Action>) => {
+  switch (action.type) {
+  case UPDATE_VOTE_MODULES: {
+    let newState = Map();
+    action.voteModules.forEach((m) => {
+      let moduleInfo = Map();
+      moduleInfo = Map({
+        type: m.type,
+        id: m.id,
+        titleEntries: fromJS(m.titleEntries),
+        instructionsEntries: fromJS(m.instructionsEntries),
+        exclusive: m.exclusive,
+        tokenTypes: List(m.tokenTypes.map(t => t.id))
+      });
+      newState = newState.set(m.id, moduleInfo);
+    });
+    return newState;
+  }
+  case CREATE_TOKEN_VOTE_MODULE:
+    return state.set(action.id, defaultTokenModule.set('id', action.id));
+  case UPDATE_TOKEN_VOTE_EXCLUSIVE:
+    return state.setIn([action.id, 'exclusive'], action.value);
+  case UPDATE_TOKEN_VOTE_INSTRUCTIONS:
+    return state.updateIn([action.id, 'instructionsEntries'], updateInLangstringEntries(action.locale, action.value));
+  case CREATE_TOKEN_VOTE_TYPE: {
+    let tokenTypes = state.getIn([action.parentId, 'tokenTypes']);
+    tokenTypes = tokenTypes.push(action.id);
+    return state.setIn([action.parentId, 'tokenTypes'], tokenTypes);
+  }
+  case DELETE_TOKEN_VOTE_TYPE: {
+    let tokenTypes = state.getIn([action.parentId, 'tokenTypes']);
+    const numberToDelete = tokenTypes.size - action.value;
+    for (let i = 0; i < numberToDelete; i += 1) {
+      tokenTypes = tokenTypes.delete(tokenTypes.size - 1);
+    }
+    return state.setIn([action.parentId, 'tokenTypes'], tokenTypes);
```

Comment:
same here `.updateIn` with `.delete`
I don't understand the code here, it seems that you remove more that one tokenType. I don't find a test, it would document the code or allow you to find the bug (if any)